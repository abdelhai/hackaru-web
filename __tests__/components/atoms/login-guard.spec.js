import { Store } from 'vuex-mock-store';
import { shallowMount } from '@vue/test-utils';
import LoginGuard from '@/components/atoms/login-guard';

describe('LoginGuard', () => {
  const $store = new Store({});
  const $router = { replace: jest.fn() };

  const factory = () =>
    shallowMount(LoginGuard, {
      mocks: {
        $store,
        $router,
        $route: { fullPath: '/secure' }
      }
    });

  beforeEach(() => {
    sessionStorage.clear();
    $store.reset();
  });

  describe('when user is not logged-in', () => {
    beforeEach(() => {
      $store.getters['auth/loggedIn'] = false;
      factory();
    });

    it('redirect to login page', () => {
      expect($router.replace).toHaveBeenCalledWith('/en/auth');
    });

    it('save current path', () => {
      expect(sessionStorage.setItem).toHaveBeenCalledWith(
        'previousPath',
        '/secure'
      );
    });
  });

  describe('when user is logged-in', () => {
    beforeEach(() => {
      $store.getters['auth/loggedIn'] = true;
      factory();
    });

    it('does not redirect', () => {
      expect($router.replace).not.toHaveBeenCalled();
    });
  });
});
