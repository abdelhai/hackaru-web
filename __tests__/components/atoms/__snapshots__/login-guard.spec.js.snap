// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginGuard when user has valid token does not redirect 1`] = `
<section>
  <p>
    Content
  </p>
</section>
`;
